### ISSUE TITLE
Road to v1.0.0

### ISSUE THREAD

**Issue by caarlos0** [goreleaser/goreleaser#807]
So, I think it is time we can start to plan v1.0.0.

What I have in mind so far is:

- [x] Support multiple archives
- [x] Clarify snapshots
- [x] Monorepo support with tag prefixes ( #723 )
- [x] split the build/packaging from the publishing phase (#816)
- [x] gitlab support ( #1038 )
- [x] wasm builds ( #1017 )

I think that once this is done and we are stable for a couple of weeks, we can cut the v1 and maybe start thinking on v2!

What do you think?

**Comment by sidh** [comment]
Monorepo support would be welcome.

**Comment by soupdiver** [comment]
Willing to help ... trying to get an overview atm

**Comment by caarlos0** [comment]
> Willing to help ... trying to get an overview atm

awesome, let me know if I can help on anything or if something can be better for new people wanting to contribute 🚀

**Comment by soupdiver** [comment]
>  let me know if I can help on anything

As said in the other PR, it's not completely obvious what is the direction right now.
A lot of issues/PRs have `won't fix`, are closed or were just experiments for features.

_If_ this ticket is still up-to-date then I think the mentioned PRs need some clarification.
#787 for example. It's still in the todo list here but the PR is closed and the branch is deleted.

So, making it easier to grasp _where_ exactly help is needed would be beneficial for new people to start contributing I think.

**Comment by robdefeo** [comment]
@caarlos0 would you consider moving https://github.com/goreleaser/goreleaser/issues/1017 to v1.0.0 milestone?

**Comment by caarlos0** [comment]
I think we can, yes.

**Comment by robdefeo** [comment]
Amazing!!! 🍰 This will help a lot we are looking to get a wasm build in place to support browser implementation.

**Comment by caarlos0** [comment]
cc/ @robdefeo https://github.com/goreleaser/goreleaser/pull/1159

**Comment by caarlos0** [comment]
wasm support is on master.

**Comment by karuppiah7890** [comment]
@caarlos0 Any updates on the mono repo support? Is anyone working on it? I see that the existing PR #723 has been closed with not much activity

**Comment by caarlos0** [comment]
> @caarlos0 Any updates on the mono repo support? Is anyone working on it? I see that the existing PR #723 has been closed with not much activity

monorepos work given that you release everything in the monorepo on all tags... otherwise, no, no one working on it afaik.

**Comment by jphastings** [comment]
What was needed for #723? I would really benefit from the monorepo tag prefix support, and could dedicate some time to taking that PR over, if there was clear direction on what needed to be done.

**Comment by caarlos0** [comment]
honestly, it got lost in the ocean of issues and PRs...

feel free to work from it and open a new pr :)

**Comment by jphastings** [comment]
@caarlos0 I started having a look at it, and I'm worried it'd be a pretty major change to `goreleaser`; right now the `Config` and `Context` structure all make the assumption that all of the builds are being released together (eg. there's only one `brews` definition, not one per buildable unit). In a monorepo I think that one commit with more than one version tag (eg. `component-a/0.1.0` and `component-c/1.2.0`) would want to trigger a build of _only_ those two components (eg. `cmd/componen-a/main.go` and `cmd/component-c/main.go`, but _not_ `component-b/main.go`) and have them released as their own Github (/Gitea/Gitlab) releases _and_ have each released as their own Homebrew Formula.

I' not yet sure I have that much time to give, but assuming I did, is goreleaser ready for such a change? Is there a design review process for discussing how this might be achieved beforehand?
